credsweeper/file_handler/text_content_provider.py

```diff
@@ -23,5 +24,13 @@ def get_analysis_target(self) -> List[AnalysisTarget]:
             list of analysis targets based on every row in file
 
         """
-        lines = Util.read_file(self.file_path)
+        if Util.get_extension(self.file_path) == ".xml":
+            try:
+                tree = ElementTree.parse(self.file_path)
+                lines = Util.get_xml_data(tree.getroot())
+            except ElementTree.ParseError:
```

Comment:
Possibly common Exception should be caught? 
There may be encoding error. Not sure it wrapped in xml parser